# YoutubeCommentsDataSet.csv exploration for fine tuning

In [1]:
import pandas as pd 
import numpy as np 

In [2]:
raw_data = pd.read_csv("https://raw.githubusercontent.com/ArrafMelon/Youtube-Video-Sentiment/main/data/raw_data/YoutubeCommentsDataSet.csv")
raw_data.head()

Comment Sentiment
0  lets not forget that apple pay in 2014 require...   neutral
1  here in nz 50 of retailers don’t even have con...  negative
2  i will forever acknowledge this channel with t...  positive
3  whenever i go to a place that doesn’t take app...  negative
4  apple pay is so convenient secure and easy to ...  positive

In [3]:
raw_data.shape

(18408, 2)

In [4]:
raw_data["Sentiment"].value_counts()

Sentiment
positive    11432
neutral      4638
negative     2338
Name: count, dtype: int64

## Data Cleaning
Getting rid of duplicates and null data, for cleaner data used for fine tuning

In [5]:
raw_data = raw_data.drop_duplicates()
raw_data.shape

(17877, 2)

In [6]:
raw_data = raw_data.dropna()
raw_data.shape

(17874, 2)

There are three classes, and 17874 rows of youtube comments data (with duplicates and nulls dropped), which should be good enough for fine-tuning use.


## Integer classes

0: Negative

1: Neutral

2: Positive

In [7]:
raw_data["Sentiment"] = raw_data["Sentiment"].replace({"negative":0,"neutral":1,"positive":2}).astype(int)

C:\Users\araf7\AppData\Local\Temp\ipykernel_8852\3478267845.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  raw_data["Sentiment"] = raw_data["Sentiment"].replace({"negative":0,"neutral":1,"positive":2}).astype(int)


In [8]:
raw_data.dtypes

Comment      object
Sentiment     int64
dtype: object

In [9]:
raw_data["Sentiment"].value_counts()

Sentiment
2    11054
1     4503
0     2317
Name: count, dtype: int64

## Save into Directory

In [10]:
raw_data.to_csv("CleanYoutubeCommentsDataSet.csv")